omigami/data/data_splitter.py

```diff
@@ -0,0 +1,89 @@
+from typing import Union, Dict, List
+
+from omigami.data import RandomState, InputDataset, Split, TrainTestData, NumpyArray
+from sklearn.model_selection import GroupShuffleSplit
+
+
+class DataSplitter:
+    def __init__(
+        self,
+        n_outer: int,
+        n_inner: int,
+        input_data: InputDataset,
+        random_state: Union[int, RandomState],
+    ):
+        self.n_outer = n_outer
+        self.n_inner = n_inner
+        self.random_state = random_state
+        self._splits = self._make_splits(input_data)
+
+    def _make_splits(self, input_data: InputDataset) -> Dict[tuple, Split]:
+        """Create a dictionary of split indexes for i`input_data`,
```

Comment:
This comment focuses too much on implementation details imho so it is q bit useless. I would put a comment here like this:
"""
Create inner and outer splits.

Creates nested splits of the input data. The outer training datasets will be split again to provide validation and inner training datasets.
"""